docs/getting_started/integration_tests.md

```diff
@@ -0,0 +1,150 @@
+# Integration tests for your API
+
+The _money-exchange_ application exposes two endpoints `euro` and `euro-at-date` which can convert an amount in a given currency to its equivalent in euros, or into its equivalent in euros using the exchange rate at a given date, respectively.
+
+To test this application, one needs to be able to issue calls to both endpoints, and check the results for correctness. Luckily Gosoline offers plenty of help with this.
+
+## Package _suite.Suite_
+
+This package acts as a wrapper over Golang's _testing_, and each _Suite_ object implements the following interface:
+
+[embedmd]:# (../../pkg/test/suite/suite.go /type TestingSuite interface/ /\n}/)
+```go
+type TestingSuite interface {
+	Env() *env.Environment
+	SetEnv(environment *env.Environment)
+	SetT(t *testing.T)
+	T() *testing.T
+	SetupSuite() []Option
+}
+```
+
+The starting point for a Gosoline integration test execution is the `suite.Run` function:
+
+[embedmd]:# (../../pkg/test/suite/run.go /func Run/ /\)/)
+```go
+func Run(t *testing.T, suite TestingSuite, extraOptions ...Option)
+```
+
+This method will use reflection to find all test cases declared by a given _TestingSuite_ object, apply the Option then create a kernel with whatever modules or APIs were declared in the `SetupSuite` and `SetupApiDefinitions`, and lastly run the tests.
+
+Each Gosoline integration test follows the same format:
+
+- Creates an object which implements _TestingSuite_
+- Implements the `SetupSuite` method for that object
+- Has at least one `Test...` method
+- It calls `suite.Run`
+
+## Practical example
+
+Inside `examples/getting_started/integration` we have an integration test for _money-exchange_. It is comprised out of two files, `api_test.go` and `fixtures.go`.
+
+`fixtures.go` contains initial values for the `currency` key value store. Its most important part is:
+
+[embedmd]:# (../../examples/getting_started/integration/fixtures.go /var fixtureSets/ /\n}/)
+```go
+var fixtureSets = []*gosoFixtures.FixtureSet{
+	{
+		Enabled: true,
+		Writer:  gosoFixtures.ConfigurableKvStoreFixtureWriterFactory("currency"),
+		Fixtures: []interface{}{
+			&gosoFixtures.KvStoreFixture{
+				Key:   "GBP",
+				Value: 1.23,
+			},
+			&gosoFixtures.KvStoreFixture{
+				Key:   "2021-01-03-GBP",
+				Value: 1.24,
+			},
+		},
+	},
+}
+```
+
+This _FixtureSet_ holds exchange rate data for GBP, which will be needed by the test.
```

Comment:
Why? Because we don't want to download exchange rates from external servers in a test, but maybe we could include more information than "we will need that for some undisclosed reason"